src/babylon.engine.ts

```diff
@@ -976,17 +987,45 @@
         }
 
         public bindArrayBuffer(buffer: WebGLBuffer): void {
-            this.bindBuffer(buffer, this._gl.ARRAY_BUFFER);
+            if (this._currentBoundBuffer[this._gl.ARRAY_BUFFER] !== buffer) {
+                this._gl.bindBuffer(this._gl.ARRAY_BUFFER, buffer);
+                this._currentBoundBuffer[this._gl.ARRAY_BUFFER] = buffer;
+            }
         }
 
         public updateArrayBuffer(data: Float32Array): void {
             this._gl.bufferSubData(this._gl.ARRAY_BUFFER, 0, data);
         }
 
-        private bindBuffer(buffer: WebGLBuffer, target: number): void {
-            if (this._currentBoundBuffer[target] !== buffer) {
-                this._gl.bindBuffer(target, buffer);
-                this._currentBoundBuffer[target] = buffer;
+        private vertexAttribPointer(buffer: WebGLBuffer, indx: number, size: number, type: number, normalized: boolean, stride: number, offset: number): void {
+            var pointer = this._currentBufferPointers[indx];
+
+            var changed = false;
+            if (!pointer) {
+                changed = true;
+                this._currentBufferPointers[indx] = { indx, size, type, normalized, stride, offset, buffer: buffer };
+            } else {
+                if (pointer.buffer !== buffer) { pointer.buffer = buffer; changed = true; }
+                if (pointer.size !== size) { pointer.size = size; changed = true; }
+                if (pointer.type !== type) { pointer.type = type; changed = true; }
+                if (pointer.normalized !== normalized) { pointer.normalized = normalized; changed = true; }
+                if (pointer.stride !== stride) { pointer.stride = stride; changed = true; }
+                if (pointer.offset !== offset) { pointer.offset = offset; changed = true; }
+            }
+
+            if (changed) {
+                if (this._currentBoundBuffer[this._gl.ARRAY_BUFFER] !== buffer) {

```

Comment:
why not keeping bindBuffer function here? It is the same code as in bibdArrayBuffer